Tests/SonarScanner.MSBuild.PreProcessor.Test/RoslynAnalyzerProviderTests.cs

```diff
@@ -230,147 +175,157 @@ private List<SonarRule> CreateRules() =>
+            CheckFileIsXml(ruleSetPath);
+            var expectedFileName = isDeactivated ? $"Sonar-{Language}-none.ruleset" : $"Sonar-{Language}.ruleset";
+            Path.GetFileName(ruleSetPath).Should().Be(expectedFileName, "Ruleset file does not have the expected name");
+            Path.GetDirectoryName(ruleSetPath).Should().Be(Path.Combine(RootDir, "conf"), "Ruleset was not written to the expected location");
+            var expectedRule = isDeactivated ? "None" : "Warning";
+            var expectedContent = $"""
+                <?xml version="1.0" encoding="utf-8"?>
+                <RuleSet xmlns:xsd="http://www.w3.org/2001/XMLSchema" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" Name="Rules for SonarQube" Description="This rule set was automatically generated from SonarQube" ToolsVersion="14.0">
+                  <Rules AnalyzerId="SonarAnalyzer.CSharp" RuleNamespace="SonarAnalyzer.CSharp">
+                    <Rule Id="S1116" Action="{expectedRule}" />
+                    <Rule Id="S1125" Action="{expectedRule}" />
+                    <Rule Id="S1000" Action="None" />
+                  </Rules>
+                  <Rules AnalyzerId="Wintellect.Analyzers" RuleNamespace="Wintellect.Analyzers">
+                    <Rule Id="Wintellect003" Action="{expectedRule}" />
+                  </Rules>
+                </RuleSet>
+                """;
+            File.ReadAllText(ruleSetPath).Should().Be(expectedContent, "Ruleset file does not have the expected content: {0}", ruleSetPath);
+        }
 
-        // Dump the contents to help with debugging
-        TestContext.AddResultFile(actualFilePath);
-        TestContext.WriteLine("File contents: {0}", actualFilePath);
-        TestContext.WriteLine(File.ReadAllText(actualFilePath));
-        TestContext.WriteLine(string.Empty);
+        private static IList<AnalyzerPlugin> CreateAnalyzerPlugins(List<string[]> pluginList) =>
+            pluginList.Select(x => new AnalyzerPlugin { AssemblyPaths = new List<string>(x) }).ToList();
 
-        if (expectedContent is not null)
+        private static void CheckFileIsXml(string fullPath)
         {
-            File.ReadAllText(actualFilePath).Should().Be(expectedContent, "Additional file does not have the expected content: {0}", expectedFileName);
+            var doc = new XmlDocument();
+            doc.Load(fullPath);
+            doc.FirstChild.Should().NotBeNull("Expecting the file to contain some valid XML");
         }
-    }
 
-    private static void CheckFileIsXml(string fullPath)
-    {
-        var doc = new XmlDocument();
-        doc.Load(fullPath);
-        doc.FirstChild.Should().NotBeNull("Expecting the file to contain some valid XML");
-    }
-
-    private static void CheckExpectedAssemblies(AnalyzerSettings actualSettings, params string[] expected)
-    {
-        foreach (var expectedItem in expected)
+        private static string CreateTestFolders(TestContext testContext)
```

Comment:
After removal of the ctors, this can use local test context
```suggestion
        private string CreateTestFolders()
```